src/database/migration/v2.js

```diff
@@ -133,8 +136,80 @@ const v2Migrations = [
     },
   },
   {
-    version: '2.3.6',
+    version: '2.3.6-IC1',
     migrate: database => {
+      // *** PART ONE, find foreign item line and delete
+
+      const itemBatches = this.database.objects('ItemBatch').snapshot();
+      const itemBatcheIDsToDelete = [];
+      // Assuming this exists everywhere
+      const inventoryAdjustmentName = database
+        .objects('Name')
+        .filtered('type = $0', 'inventory_adjustment')[0];
+
+      itemBatches.forEach(({ id, transactionBatches }) => {
+        // No transaction batches on item line should mean it's foreign
+        if (transactionBatches.length === 0) {
+          itemBatcheIDsToDelete.push(id);
+          return;
+        }
+        // supplier_invoice type transactions are all addition transactions
+        const additionTransactionBatches = transactionBatches.filtered(
+          'transaction.type = "supplier_invoice"'
+        );
+        // First try to find supplier invoice, if found this must be a local ItemBatch
+        const supplierInvoiceBatches = additionTransactionBatches.filtered(
+          'transaction.otherParty.id != $0',
+          inventoryAdjustmentName.id
+        );
+        if (supplierInvoiceBatches.length > 0) return;
+
+        // Find all 'addition' inventory adjustments
+        const additionInventoryAdjustemnetBatches = additionTransactionBatches.filtered(
```

Comment:
`const additionInventoryAdjustmentBatches` hehe also this is the same query as for supplier invoices above? I think this is meant to be just for addition transaction batches not specifically

Reply:
im stupid